endpoints/osruntime-bootstrap

```diff
@@ -0,0 +1,245 @@
+#
+# osruntime-bootstrap: To get a rickshaw engine running:
+# 1) Create a new script, by declaring the below variables
+# 2) Add the rest of this script to the new script
+# The above can be done with make_osruntime_boostrap_script() in ./base
+
+# Vars which should be declared when duplicating this script:
+# (If this list changes, assignment in load_osruntime also had to change)
+required_vars=""
+required_vars+=" run_id"
+required_vars+=" rickshaw_host"
+required_vars+=" endpoint_run_dir"
+required_vars+=" cs_label"
+required_vars+=" base_run_dir"
+required_vars+=" cpu_partitioning"
+required_vars+=" endpoint"
+required_vars+=" osruntime"
+required_vars+=" max_sample_failures"
+required_vars+=" max_rb_attempts"
+required_vars+=" rb_passwd"
+required_vars+=" rb_id"
+required_vars+=" ssh_id"
+required_vars+=" image"
+required_vars+=" disable_tools"
+
+base_dir="/var/lib/crucible"
+cfg_dir="${base_dir}/cfg"
+data_dir="${base_dir}/data/tmp"
+logs_dir="${base_dir}/logs"
+mkdir -p $cfg_dir
+mkdir -p $data_dir
+mkdir -p $logs_dir
+log_file_name="osruntime-bootstrap-logfile_${run_id}_$cs_label.txt"
+log_file="$logs_dir/$log_file_name"
+remote_log_file="$rickshaw_host:$endpoint_run_dir/$log_file_name"
+ssh_opts="-o StrictHostKeyChecking=no -o PasswordAuthentication=no"
+ssh_id_dir="/root/.ssh"
+ssh_id_file="/tmp/ssh-id_${run_id}_$cs_label.id"
+ssh_pub_id_file="$ssh_id_dir/id_rsa.pub"
+if [ ! -z "$ssh_id" ]; then
+    mkdir -p $ssh_id_dir
+    chmod 700 $ssh_id_dir
+    echo -e $ssh_id >$ssh_id_file && chmod 600 $ssh_id_file
+    echo "$ssh_id_dir: $ssh_id_dir"
+    /bin/ls -la $ssh_id_dir
+    echo "$ssh_id_file:"
+    cat $ssh_id_file
+else
+    echo "ssh_id not defined, cannot continue"
+    echo "$ssh_id"
+    exit 1
+fi
+
+function log_output() {
+    echo -e "$@"
+    echo -e "$@" >>$log_file
+}
+
+function exit_error() {
+    # Note that if ssh_id, rickshaw_host, and endpoint_run_dir are not defined, scp will not work
+    log_output $@
+    log_output "Going to scp with: scp $ssh_opts -i $ssh_id_file $log_file $remote_log_file"
+    scp $ssh_opts -i $ssh_id_file $log_file $remote_log_file
+    exit 1
+}
+
+function exit_success() {
+    # Note that if ssh_id, rickshaw_host, and endpoint_run_dir are not defined, scp will not work
+    log_output $@
+    log_output "Going to scp with: scp $ssh_opts -i $ssh_id_file $log_file $remote_log_file"
+    scp $ssh_opts -i $ssh_id_file $log_file $remote_log_file
+    exit 0
+}
+
+function validate_vars() {
+    log_output "validate_vars() begin"
+    # Fail if any required vars are missing
+    for this_var in $required_vars; do
+        if [[ -z  "${!this_var}" ]]; then
+            exit_error "variable $this_var is not initialized"
+        fi
+    done
+    log_output "validate_vars() end"
+}
+
+validate_vars
+
+image_cache_size=3
+chroot_rbind_mounts="proc dev sys lib/modules usr/src boot var/run"
+host_mounts="run/openvswitch"
```

Comment:
this seems odd to me...what is OVS isn't present?

Reply:
The osruntime-bootstrap checks for existence of the directory in the host first, and will not create the mount if it does not exist.